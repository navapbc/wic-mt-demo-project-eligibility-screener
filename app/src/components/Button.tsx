import Image from 'next/image'
import Link from 'next/link'
import styled, { StyledComponent } from 'styled-components'

type Props = {
  text: string
  href: string /* TODO: create global type for routes */
  vector?: boolean
  width?: string /* TODO: check if there is a type for css widths */
}

const Button = (props: Props) => {
  const { width, text, vector, href } = props

  return (
    <Link href={href}>
      <Btn className="usa-button usa-button--small" width={width}>
        {text}
        {vector && (
          <Image
            src="/img/vector.svg"
            alt="continue vector"
            width={20}
            height={15}
          />
        )}
      </Btn>
    </Link>
  )
}

const Btn: StyledComponent<'a', object, { width?: string }, never> = styled.a<{
  width?: string
}>`
  background-color: black;
  display: flex;
  font-family: 'Balsamiq Sans', cursive;
  font-weight: 400;
  gap: 10px;
  width: ${(props) => (props.width ? props.width : '90%')};
`

export default Button
