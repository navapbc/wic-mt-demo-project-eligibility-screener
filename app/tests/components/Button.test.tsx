import { render, screen } from '@testing-library/react'

import Button, { buttonStyleOptions } from '@components/Button'

const styleOptions = buttonStyleOptions.map((style) => [style])

it.each(styleOptions)('should match snapshot for %s style', (style) => {
  const { container } = render(
    <Button disabled={false} labelKey="button label" style={style} />
  )
  expect(container).toMatchSnapshot()
})

it.each(styleOptions)(
  'should have the correct css class for %s style',
  (style) => {
    render(<Button labelKey="button label" style={style} />)
    const button = screen.getByRole('button')

    if (style !== 'default') {
      expect(button).toHaveClass(`usa-button--${style}`)
    }
  }
)

it('should be disabled if disabled is true', () => {
  render(<Button disabled={true} labelKey="button label" />)
  const button = screen.getByRole('button')
  expect(button).toBeDisabled()
})

it('should be disabled if disabled is false', () => {
  render(<Button disabled={false} labelKey="button label" />)
  const button = screen.getByRole('button')
  expect(button).not.toBeDisabled()
})
