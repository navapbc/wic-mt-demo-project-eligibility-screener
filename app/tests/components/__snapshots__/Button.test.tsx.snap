// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for accent-cool style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block usa-button--accent-cool margin-top-1"
  >
    button label
  </button>
</div>
`;

exports[`should match snapshot for accent-warm style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block usa-button--accent-warm margin-top-1"
  >
    button label
  </button>
</div>
`;

exports[`should match snapshot for base style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block usa-button--base margin-top-1"
  >
    button label
  </button>
</div>
`;

exports[`should match snapshot for big style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block usa-button--big margin-top-1"
  >
    button label
  </button>
</div>
`;

exports[`should match snapshot for default style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block margin-top-6"
  >
    button label
  </button>
</div>
`;

exports[`should match snapshot for outline style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block usa-button--outline margin-top-1"
  >
    button label
  </button>
</div>
`;

exports[`should match snapshot for secondary style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block usa-button--secondary margin-top-1"
  >
    button label
  </button>
</div>
`;

exports[`should match snapshot for unstyled style 1`] = `
<div>
  <button
    class="usa-button usa-button--small display-block usa-button--unstyled margin-top-1"
  >
    button label
  </button>
</div>
`;
